packages/manager/apps/dedicated/client/app/dedicated/server/dashboard/technical-details/technical-details.controller.js

```diff
@@ -0,0 +1,81 @@
+import filter from 'lodash/filter';
+import get from 'lodash/get';
+
+export default class TechnicalDetailsController {
+  /* @ngInject */
+  constructor($translate) {
+    this.$translate = $translate;
+  }
+
+  $onInit() {
+    this.formattedCPU = this.formatCPU();
+    this.formattedRAM = this.formatRAM();
+    this.formattedDataDisks = this.formatDisks('data');
+    this.formattedCacheDisks = this.formatDisks('cache');
+    this.formattedOsDisks = this.formatDisks('os');
+    this.formattedExtensionCard = get(
+      this.technicalDetails,
+      'storage.raid',
+      '-',
+    );
+  }
+
+  formatRAM() {
+    const ram = get(this.technicalDetails, 'memory');
+    if (!ram) {
+      return '-';
+    }
+    const freqUnit = this.$translate.instant(
+      'dedicated_server_dashboard_technical_details_ram_frequency_unit',
+    );
+    const ramSize = ram.size ? `${ram.size} GB` : '';
+    const ramType = get(ram, 'type', '');
+    const ramECC = ram.ecc ? 'ECC' : '';
+    const ramFrequency = ram.frequency ? `${ram.frequency} ${freqUnit}` : '';
+    return `${ramSize} ${ramType} ${ramECC} ${ramFrequency}`;
+  }
+
+  formatCPU() {
+    const cpu = get(this.technicalDetails, 'server.cpu');
+    if (!cpu) {
+      return '-';
+    }
+    const freqUnit = this.$translate.instant(
+      'dedicated_server_dashboard_technical_details_cpu_frequency_unit',
+    );
+    const cpuBrand = get(cpu, 'brand', '');
+    const cpuModel = get(cpu, 'model', '');
+    const cpuCores = cpu.cores ? `${cpu.cores}c` : '';
+    const cpuThreads = cpu.threads ? `${cpu.cores && '/'}${cpu.threads}t` : '';
+    const cpuFrequency = cpu.frequency ? `${cpu.frequency} ${freqUnit}` : '';
+    const cpuBoost =
+      cpu.boost !== cpu.frequency ? `/${cpu.boost} ${freqUnit}` : '';
+    return `${cpuBrand} ${cpuModel} - ${cpuCores}${cpuThreads} - ${cpuFrequency}${cpuBoost}`;
+  }
+
+  formatDisks(diskUsage) {
+    const disks = filter(
+      get(this.technicalDetails, 'storage.disks'),
+      (disk) => disk.usage === diskUsage,
+    );
+    if (!disks.length) {
+      return ['-'];
+    }
+    return disks.map((disk) => {
+      const number = get(disk, 'number', 1);
+      const technology = get(disk, 'technology', '');
+      const interf = get(disk, 'interface', '');
+      let capacity = Number(get(disk, 'capacity'));
+
+      if (Number.isNaN(capacity)) {
+        capacity = '-';
+      } else if (capacity >= 1000) {
+        capacity = `${Math.round((100 * capacity) / 1000, 2) / 100} TB`;
```

Comment:
Maybe this module could help you ? https://github.com/ovh/manager/blob/master/packages/manager/modules/filters/src/index.js